addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java

```diff
@@ -201,10 +205,14 @@ public IpPingMethodEnum determinePingMethod() {
      */
     public ArpPingUtilEnum determineNativeARPpingMethod(String arpToolPath) {
         String result = ExecUtil.executeCommandLineAndWaitResponse(arpToolPath + " --help", 100);
+        logger.trace("Determining arping tool: {}", arpToolPath);
```

Comment:
It would be far more useful if you are adding tracing to add it to the detection success or failure each time it tries to detect. That's the approach I took in my testing. So if the arping returns true, I log a FOUND message for that hostname and then added an else block to log NOT FOUND. That allows me to get up in the morning and parse the logs to see my success/failure rate far more easily (I could probably use persistence or something but that's more effort than I want to put into it). Logging that we are checking or determining the tool isn't super useful past initial setup which can just be seen in the properties, as noted.

Reply:
Hm. I'm sorry @longwa, but because you are too lazy to setup a persistence service (one click in paper ui and 5 config lines), everybody should live with a performance penalty? I don't agree.

Reply:
A performance penalty for a trace log? Hahaha. So you are one of those developers that thinks a disabled log message is too much overhead? I’m sure you are joy to work with day to day. This plugin is a hot mess anyway so feel free to do whatever you want. I built a version that actually works but given  your presumptive and condescending attitude toward my comment I see no reason to open a pull request. 

I do have persistence configured but I only need it for restart so I use mapdb. All I wanted for this plugin is a little more feedback since a quick google search shows dozens of confused people about how and whether it’s working. I guess since log statements are too expensive we all have to live with actually costly persistence for what should be a log call. No thanks.

Reply:
A c++ disabled log message is not linked. We do not have this luxury in java. In every log message call the runtime is asked for the current logging state (all function parameters are resolved by the way, so if they are expensive, they still happen) and a comparison to the requested logging level is happening. 
Because of people like you Java has the image of being slow. Thanks.

Reply:
I’ve been working in Java for over 20 years and am well aware of how logging works. Unless the logging is in a tight loop, any performance penalty is so insignificant that it is almost always overshadowed by any number of things. In my scenario, switching from mapdb to MySQL for persistence would probably have 10x more impact on performance than having a few logging statements. If you are going to work on something open source you might try being less of an asshole when people offer their time for something. Good luck.